Source/ServerTrustEvaluation.swift

```diff
@@ -188,47 +221,65 @@ public final class RevocationTrustEvaluator: ServerTrustEvaluating {
+    ///                                   to performing the default evaluation, even if `performDefaultValidation` is
+    ///                                   `false`. Defaults to `true`.
     public init(certificates: [SecCertificate] = Bundle.main.certificates,
-                validateCertificateChain: Bool = true,
+                acceptSelfSignedCertificates: Bool = false,
+                performDefaultValidation: Bool = true,
                 validateHost: Bool = true) {
         self.certificates = certificates
-        self.validateCertificateChain = validateCertificateChain
+        self.acceptSelfSignedCertificates = acceptSelfSignedCertificates
+        self.performDefaultValidation = performDefaultValidation
         self.validateHost = validateHost
     }
 
-    /// Evaluates the given `SecTrust` value for the given `host`.
-    ///
-    /// - Parameters:
-    ///   - trust: The `SecTrust` value to evaluate.
-    ///   - host:  The host for which to evaluate the `SecTrust` value.
-    /// - Returns: Whether or not the evaluator considers the `SecTrust` value valid for `host`.
-    public func evaluate(_ trust: SecTrust, forHost host: String) -> Bool {
-        if validateCertificateChain {
-            let policy = SecPolicyCreateSSL(true, validateHost ? host as CFString : nil)
-            SecTrustSetPolicies(trust, policy)
-
-            SecTrustSetAnchorCertificates(trust, certificates as CFArray)
-            SecTrustSetAnchorCertificatesOnly(trust, true)
-
-            return trust.isValid
-        } else {
-            let serverCertificatesData = Set(trust.certificateData)
-            let pinnedCertificatesData = Set(certificates.data)
-
-            return !serverCertificatesData.isDisjoint(with: pinnedCertificatesData)
+    public func evaluate(_ trust: SecTrust, forHost host: String) throws -> Bool {
+        guard !certificates.isEmpty else {
+            throw AFError.serverTrustEvaluationFailed(reason: .noCertificatesFound)
         }
+        
+        if acceptSelfSignedCertificates {
+            try trust.setAnchorCertificates(certificates)
+        }
+        
+        if performDefaultValidation {
+            try trust.validate(policy: .default) { (status, result) in
+                AFError.serverTrustEvaluationFailed(reason: .defaultEvaluationFailed(output: .init(host, trust, status, result)))
+            }
+        }
+        
+        if validateHost {
+            try trust.validate(policy: .hostname(host)) { (status, result) in
+                AFError.serverTrustEvaluationFailed(reason: .hostValidationFailed(output: .init(host, trust, status, result)))
+            }
+        }
+        
+        let serverCertificatesData = Set(trust.certificateData)
+        let pinnedCertificatesData = Set(certificates.data)
+        let certificatesPinned = !serverCertificatesData.isDisjoint(with: pinnedCertificatesData)
+        if !certificatesPinned {
```

Comment:
Might be more clear to refactor `certificatesPinned` to `pinnedCertificatesFound` or `pinnedCertificateFoundInServerCertificates` or `pinnedCertFoundInServerCerts`.

Reply:
Updated naming.